jumpcloud-ADMU/Powershell/Start-Migration.ps1

```diff
@@ -1455,18 +1855,102 @@ Function Start-Migration {
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][bool]$AutobindJCUser = $false,
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][bool]$BindAsAdmin = $false,
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][bool]$SetDefaultWindowsUser = $true,
+        [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][bool]$AdminDebug = $false,
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][ValidateLength(40, 40)][string]$JumpCloudConnectKey,
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][ValidateLength(40, 40)][string]$JumpCloudAPIKey,
         [Parameter(ParameterSetName = 'cmd', Mandatory = $false)][ValidateLength(24, 24)][string]$JumpCloudOrgID,
         [Parameter(ParameterSetName = "form")][Object]$inputObject)
 
     Begin {
+        # Define misc static variables
+        $netBiosName = Get-NetBiosName
+        $WmiComputerSystem = Get-WmiObject -Class:('Win32_ComputerSystem')
+        $localComputerName = $WmiComputerSystem.Name
+        $systemVersion = Get-ComputerInfo | Select-Object OSName, OSVersion, OsHardwareAbstractionLayer
+        $windowsDrive = Get-WindowsDrive
+        $jcAdmuTempPath = "$windowsDrive\Windows\Temp\JCADMU\"
+        $jcAdmuLogFile = "$windowsDrive\Windows\Temp\jcAdmu.log"
+        $netBiosName = Get-NetBiosName
+
+        # JumpCloud Agent Installation Variables
+        $AGENT_PATH = Join-Path ${env:ProgramFiles} "JumpCloud"
+        $AGENT_BINARY_NAME = "jumpcloud-agent.exe"
+        $AGENT_INSTALLER_URL = "https://cdn02.jumpcloud.com/production/jcagent-msi-signed.msi"
+        $AGENT_INSTALLER_PATH = "$windowsDrive\windows\Temp\JCADMU\jcagent-msi-signed.msi"
+        $AGENT_CONF_PATH = "$($AGENT_PATH)\Plugins\Contrib\jcagent.conf"
+        $admuVersion = '2.7.0'
+
+        $script:AdminDebug = $AdminDebug
```

Comment:
We are defining AdminDebug here and outside the function, I feel like it should be defined within the start-migraiton function entirely

Reply:
$script:AdminDebug is used for the `Write-ToLog` function which is why it is set as `$script` param. We can also make it a param for `Write-ToLog` instead